rust/src/adapter/datadog.rs

```diff
@@ -22,9 +22,9 @@ use super::{
 pub struct DatadogAdapterContainer(Arc<Mutex<DatadogAdapter>>);
 
 impl DatadogAdapterContainer {
-    pub async fn start<T: 'static>(&self, linker: &mut Linker<T>) -> Result<DatadogTraceCtx> {
+    pub async fn start<T: 'static>(&self, linker: &mut Linker<T>, data: &Vec<u8>) -> Result<DatadogTraceCtx> {
```

Comment:
```suggestion
    pub async fn start<T: 'static>(&self, linker: &mut Linker<T>, data: &[u8]) -> Result<DatadogTraceCtx> {
```

I don't _think_ the reference to `Vec` reallocates, but I believe passing a reference as `&data` as input is compatible with typing this argument as `&[u8]`.